package/kedro_viz/models/graph.py

```diff
@@ -0,0 +1,373 @@
+        Returns:
+            An instance of DataNode.
+        """
+        return DataNode(
+            id=cls._hash(full_name),
+            name=_pretty_name(full_name),
+            full_name=full_name,
+            tags=tags,
+            layer=layer,
+            pipelines=[],
+            kedro_obj=dataset,
+        )
+
+    @classmethod
+    def create_parameters_node(
+        cls,
+        full_name: str,
+        layer: Optional[str],
+        tags: Set[str],
+        parameters: AbstractDataSet,
+    ) -> "ParametersNode":
+        """Create a graph node of type PARAMETERS for a given Kedro parameters dataset instance.
+
+        Args:
+            full_name: The fullname of the dataset, including namespace, e.g.
+                data_science.test_split_ratio
+            layer: The optional layer that the parameters belong to.
+            tags: The set of tags assigned to assign to the graph representation
+                of this dataset. N.B. currently it's derived from the node's tags.
+            parameters: A parameters dataset in a Kedro pipeline.
+        Returns:
+            An instance of Parameters.
+        """
+        return ParametersNode(
+            id=cls._hash(full_name),
+            name=_pretty_name(full_name),
+            full_name=full_name,
+            tags=tags,
+            layer=layer,
+            pipelines=[],
+            kedro_obj=parameters,
+        )
+
+    def belong_to_pipeline(self, pipeline_id: str) -> bool:
+        """Check whether this graph node belongs to a given pipeline_id."""
+        return pipeline_id in self.pipelines
+
+    def add_pipeline(self, pipeline_id: str):
+        """Add a pipeline_id to the list of pipelines that this node belongs to."""
+        if pipeline_id not in self.pipelines:
+            self.pipelines.append(pipeline_id)
+
+    def have_metadata(self) -> bool:
+        """Check whether this graph node has metadata.
+        Since metadata of a graph node is derived from the underlying Kedro object,
+        we just need to check whether the underlying object exists.
+        """
+        return self.kedro_obj is not None
+
+
+@dataclass
+class GraphNodeMetadata(abc.ABC):
+    """Abstract class to represents a graph node's metadata"""
+
+
+@dataclass
+class TaskNode(GraphNode):
+    """Represent a graph node of type TASK"""
+
+    kedro_obj: InitVar[KedroNode]
+    modular_pipelines: List[str] = field(init=False)
+    parameters: Dict = field(init=False, default_factory=dict)
+    type: str = GraphNodeType.TASK.value
+
+    def __post_init__(self, kedro_obj: KedroNode):
+        self._kedro_obj = kedro_obj
+
+        # the modular pipelines that a task node belongs to are derived from its namespace.
+        self.modular_pipelines = self._expand_namespaces(kedro_obj.namespace)
+
+
+@dataclass
+class TaskNodeMetadata(GraphNodeMetadata):
+    """Represent the metadata of a TaskNode"""
+
+    code: str = field(init=False)

```

Comment:
I think it would be worth adding docstring to all these classes to explain the fields. It's quite hard to figure out what `code` is if you're not familiar with the codebase.